livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -988,42 +997,62 @@ class ChannelControllerImpl(
 
     suspend fun editMessage(message: Message): Result<Message> {
         val online = domainImpl.isOnline()
-        message.updatedAt = Date()
-        message.syncStatus = SyncStatus.IN_PROGRESS
+        val now = Date()
+        var editedMessage = message.copy()
+
+        // set message.updated at if it's null or older than now (prevents issues with incorrect clocks)
+        if (editedMessage.updatedAt == null) {
+            editedMessage.updatedAt = now
+        }
+        editedMessage.updatedAt?.let {
+            if (it.before(now)) {
+                editedMessage.updatedAt = now
+            }
+        }
```

Comment:
clean syntax, editor tells me it needs a !!
bit weird since the if statement already checks for null 

```
editedMessage.apply {
            val now = Date()
            if (updatedAt == null || updatedAt!!.before(now)) {
                updatedAt = now
            }
        }
```